api/src/apps/apps.rs

```diff
@@ -258,34 +275,47 @@ impl AppsService {
         deployment_unit.extend_with_templating_only_service_configs(configs_for_templating);
 
         let images = deployment_unit.images();
-        let port_mappings = runtime.block_on(ImagesService::new().resolve_image_ports(&images))?;
+        let port_mappings = ImagesService::new().resolve_image_ports(&images).await?;
         deployment_unit.assign_port_mappings(&port_mappings);
 
         let configs: Vec<_> = deployment_unit.try_into()?;
-        let services = runtime.block_on(self.infrastructure.deploy_services(
-            app_name,
-            &configs,
-            &self.config.container_config(),
-        ))?;
+        let services = self
+            .infrastructure
+            .deploy_services(
+                &deployment_id.to_string(),
+                app_name,
+                &configs,
+                &self.config.container_config(),
+            )
+            .await?;
 
         Ok(services)
     }
 
     /// Deletes all services for the given `app_name`.
-    pub fn delete_app(&self, app_name: &AppName) -> Result<Vec<Service>, AppsServiceError> {
+    pub async fn delete_app(
+        &self,
+        app_name: &AppName,
+        deployment_id: &TaskId,
+    ) -> Result<Vec<Service>, AppsServiceError> {
         let guard = self.create_or_get_app_guard(app_name.clone(), AppGuardKind::Deletion)?;
```

Comment:
I'm wondering if we can remove the guard completely because the infrastructure now handles this with `TaskId`. Therefore, the test `should_delete_apps_from_parallel_threads_returning_the_same_result` could be restructured

Reply:
We can postpone this for another PR.

Reply:
We have #40 for this.